src/test/gc.test.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {suite} from 'uvu';
+import * as assert from 'uvu/assert';
+import {timeout} from './util/uvu-timeout.js';
+import {WireitTestRig} from './util/test-rig.js';
+import {Executor, registerExecutorConstructorHook} from '../executor.js';
+import {Analyzer} from '../analyzer.js';
+import {DefaultLogger} from '../logging/default-logger.js';
+import {WorkerPool} from '../util/worker-pool.js';
+import {registerExecutionConstructorHook} from '../execution/base.js';
+import {Deferred} from '../util/deferred.js';
+
+const test = suite<{rig: WireitTestRig}>();
+
+let numLiveExecutors = 0;
+let numLiveExecutions = 0;
+
+test.before.each(async (ctx) => {
+  try {
+    const executorFinalizationRegistry = new FinalizationRegistry(() => {
+      numLiveExecutors--;
+    });
+    registerExecutorConstructorHook((executor) => {
+      numLiveExecutors++;
+      executorFinalizationRegistry.register(executor, null);
+    });
+
+    const executionFinalizationRegistry = new FinalizationRegistry(() => {
+      numLiveExecutions--;
+    });
+    registerExecutionConstructorHook((execution) => {
+      numLiveExecutions++;
+      executionFinalizationRegistry.register(execution, null);
+    });
+    ctx.rig = new WireitTestRig();
+    await ctx.rig.setup();
+  } catch (error) {
+    // Uvu has a bug where it silently ignores failures in before and after,
+    // see https://github.com/lukeed/uvu/issues/191.
+    console.error('uvu before error', error);
+    process.exit(1);
+  }
+});
+
+test.after.each(async (ctx) => {
+  try {
+    numLiveExecutors = 0;
+    numLiveExecutions = 0;
+    await ctx.rig.cleanup();
+  } catch (error) {
+    // Uvu has a bug where it silently ignores failures in before and after,
+    // see https://github.com/lukeed/uvu/issues/191.
+    console.error('uvu after error', error);
+    process.exit(1);
+  }
+});
+
+async function retryWithGcUntilCallbackDoesNotThrow(
+  cb: () => void
+): Promise<void> {
+  for (const wait of [0, 10, 100, 500, 1000]) {
+    global.gc();
+    try {
+      cb();
+      return;
+    } catch {
+      // Ignore
+    }
+    await new Promise((resolve) => setTimeout(resolve, wait));
+  }
+  cb();
```

Comment:
why this extra call?

Reply:
Added a comment. It's the attempt after the final wait, where we don't wrap in a `try` so that the exception bubbles up.